src/pages/TodosPage.jsx

```diff
@@ -0,0 +1,89 @@
+import React, { useEffect } from 'react';
+import { useSearchParams, useNavigate } from 'react-router';
+import TodoForm from '../features/TodoForm';
+import TodoList from '../features/TodoList/TodoList';
+import TodosViewForm from '../features/TodosViewForm';
+
+function TodosPage({
+  addTodo,
+  completeTodo,
+  updateTodo,
+  sortDirection,
+  setSortDirection,
+  sortField,
+  setSortField,
+  queryString,
+  setQueryString,
+  todoState,
+}) {
+  const [searchParams, setSearchParams] = useSearchParams();
+  const navigate = useNavigate();
+
+  const itemsPerPage = 15;
+  const currentPage = parseInt(searchParams.get('page') || '1', 10);
+
+  const indexOfFirstTodo = (currentPage - 1) * itemsPerPage;
+  const totalPages = Math.ceil(todoState.todoList.length / itemsPerPage);
+
+  function handlePreviousPage() {
+    const prevPage = Math.max(currentPage - 1, 1);
+    setSearchParams({ page: prevPage.toString() });
+  }
+
+  function handleNextPage() {
+    const nextPage = Math.min(currentPage + 1, totalPages);
+    setSearchParams({ page: nextPage.toString() });
+  }
+
+  useEffect(() => {
+    if (totalPages > 0) {
+      if (
+        isNaN(currentPage) ||
+        currentPage < 1 ||
+        currentPage > totalPages
+      ) {
+        navigate('/');
+      }
+    }
+  }, [currentPage, totalPages, navigate]);
+
+  return (
+    <>
+      <TodoForm onAddTodo={addTodo} isSaving={todoState.isSaving} />
+
+      <TodoList
+        todoList={todoState.todoList.slice(indexOfFirstTodo, indexOfFirstTodo + itemsPerPage)}
+        isLoading={todoState.isLoading}
+        onCompleteTodo={completeTodo}
+        onUpdateTodo={updateTodo}
+      />
+
+      <div className="paginationControls" style={{ display: 'flex', alignItems: 'center', gap: '10px' }}>
+        <button onClick={handlePreviousPage} disabled={currentPage === 1}>
+          Previous
+        </button>
+
+        <span>
+          Page {currentPage} of {totalPages}
```

Comment:
This isn't required in the assignment at all, but just something I noticed. When your app first loads, it doesn't know the value of `totalPages` yet, so this appears like "Page 1 of 0". If you want to make it look a bit more user-friendly, you could do this:

`  Page {currentPage} of {totalPages ? totalPages : currentPage}`

This would show like "Page 1 of 1", "Page 2 of 2", etc, until `totalPages` loads, then it would update to show your true total number of pages.